src/services/completions.ts

```diff
@@ -3053,6 +3049,18 @@ namespace ts.Completions {
         }
     }
 
+    function isApparentType(type: Type, node: ObjectLiteralExpression | JsxAttributes, checker: TypeChecker) {
+        return !(type.flags & TypeFlags.Primitive
+            || checker.isArrayLikeType(type)
+            || checker.isTypeInvalidDueToUnionDiscriminant(type, node)
+            || typeHasCallOrConstructSignatures(type, checker)
+            || type.isClass() && containsHiddenProps(type.getApparentProperties()));
+    }
+
+    function containsHiddenProps(props: Symbol[]) {

```

Comment:
I'd prefer the name `containsNonPublicProperties` since I don't believe we use the term hidden anywhere else.